adapters/nanointeractive/nanointeractive.go

```diff
@@ -0,0 +1,199 @@
+package nanointeractive
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type NanointeractiveAdapter struct {
+	endpoint string
+}
+
+// used for cookies and such
+func (a *NanointeractiveAdapter) Name() string {
+	return "Nano"
+}
+
+func (a *NanointeractiveAdapter) SkipNoCookies() bool {
+	return false
+}
+
+func (a *NanointeractiveAdapter) MakeRequests(bidRequest *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	var errs []error
+	var validImps []openrtb.Imp
+
+	var adapterRequests []*adapters.RequestData
+
+	for i := 0; i < len(bidRequest.Imp); i++ {
+		err := checkImp(&bidRequest.Imp[i])
+		// If the parsing is failed, remove imp and add the error.
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		validImps = append(validImps, bidRequest.Imp[i])
+	}
+
+	if len(validImps) == 0 {
+		errs = append(errs, fmt.Errorf("no impressions in the bid request"))
+		return nil, errs
+	}
+
+	// set referrer origin
+	if refO := getRefererOrigin(&bidRequest.Imp[0]); refO != "" {
+		if bidRequest.Site == nil {
+			bidRequest.Site = &openrtb.Site{}
+		}
+		bidRequest.Site.Ref = refO
+	}
+
+	bidRequest.Imp = validImps
+
+	reqJSON, err := json.Marshal(bidRequest)
+	if err != nil {
+		errs = append(errs, err)
```

Comment:
I think it makes sense to return here. If the `reqJSON` is invalid I don't see why you'd want to proceed with the bid.

Reply:
the change is done, thanks